### PR TITLE
Export data

### PR DESCRIPTION
**Pull request by BrandonPessman**
This adds exporting data as well as bug catching for certain cases. Also changes everything to epoch time.

### CODE REVIEW

File: frontend/src/components/Export.js [hunk 1]
```diff
@@ -20,28 +21,68 @@ class Export extends Component {
...
   }
 
   componentDidUpdate() {
     var csvData = []
-    //console.log(this.state.data.data.query.electricity.data[0])
-    var data = this.state.data.data.query.electricity.data
-    csvData[0] = ['timestamp', 'value']
+    var data
+
+    if (this.state.energyType === 'electricity') {
```
Comment (alex-stout) [inline frontend/src/components/Export.js]: Solar?

File: frontend/src/components/Export.js [hunk 2]
```diff
@@ -72,6 +113,32 @@ class Export extends Component {
               >
                 <option value="Davies">Davies Student Center</option>
                 <option value="Library">McIntyre Library</option>
+                <option value="Bridgman">Bridgman</option>
```
Comment (alex-stout) [inline frontend/src/components/Export.js]: These really should be created from an API call. Not hard-coded.

File: frontend/src/components/Export.js [hunk 3]
```diff
@@ -91,6 +158,8 @@ class Export extends Component {
               >
                 <option value="electricity">Electricity</option>
```
Comment (alex-stout) [inline frontend/src/components/Export.js]: Should be created dynamically based on the building chosen.

File: frontend/src/components/GraphNavigation.js [hunk 4]
```diff
@@ -231,9 +202,9 @@ class GraphNavigation extends Component {
                   </DropdownItem>
                   <DropdownItem
```
Comment (alex-stout) [inline frontend/src/components/GraphNavigation.js]: All of these should be dynamic as well.

File: frontend/src/components/ScatterPlot.js [hunk 5]
```diff
@@ -34,101 +52,294 @@ class ScatterPlot extends Component {
...
 
   componentDidUpdate() {
+    if (this.state.populateEnergyButtons) {
+      console.log(this.state.energyTypes)
+      for (
+        let i = 0;
+        i < this.state.energyTypes.data.query.energyAvailable.length;
+        i++
+      ) {
+        console.log('TEST')
```
Comment (alex-stout) [inline frontend/src/components/ScatterPlot.js]: Remove.

File: frontend/src/components/Export.js (hunk 2 above)
Comment (BrandonPessman) [reply frontend/src/components/Export.js]: For the buildings, I think I am still waiting on a table with all the current building names if there isn't one already.

**Comment by BrandonPessman** [comment]
I am going to change it so all you see initially is the building button, then as you select a building, the next button appears, and so on until you reach the export button. Should make it flow a little better as well.

**Comment by alex-stout** [comment]
That sounds like a good plan. Do you want me to wait to merge until that is done?

**Comment by BrandonPessman** [comment]
Merging this old PR and I will be creating multiple issues to make this more efficient.

**Pull request merged**